src/SyncDb.ts

```diff
@@ -20,35 +24,93 @@ class SyncDb extends Command {
     'generate-connections': flags.boolean({ char: 'c', description: 'Generate connections' })
   };
 
+  /**
+   * Default CLI options for running synchronize.
+   *
+   * @param {*} userParams
+   * @returns {SyncParams}
+   */
+  getSyncParams(userParams: any): SyncParams {
+    return {
+      ...userParams,
+      // Individual success handler
+      onSuccess: (context: ExecutionContext) =>
+        printLine(`  [✓] ${context.connectionId} - Successful (${context.timeElapsed.toFixed(2)}s)`),
+
+      // Individual error handler
+      onFailed: (context: ExecutionContext) =>
+        printLine(`  [✖] ${context.connectionId} - Failed (${context.timeElapsed.toFixed(2)}s)`)
+    };
+  }
+
   /**
    * CLI command execution handler.
    *
    * @returns {Promise<void>}
    */
   async run(): Promise<void> {
+    // Set CLI environment as true.
+    process.env.SYNC_DB_CLI = 'true';
+
     const { flags: parsedFlags } = this.parse(SyncDb);
-    const params = {
-      force: parsedFlags.force
-    };
+    const params = this.getSyncParams({ ...parsedFlags });
 
     try {
-      await handleFlags(parsedFlags, {
-        info: this.log,
-        error: this.error
-      });
+      await handleFlags(parsedFlags);
+
       const config = await loadConfig();
       const connections = await resolveConnections();
 
-      const { synchronize } = await import('./migrator');
+      const { synchronize } = await import('./api');
+
+      await printLine('Synchronizing...\n');
 
-      await synchronize(config, connections, params);
-      process.exit(0);
+      const result = await synchronize(config, connections, params);
+
+      await this.displayResult(result);
     } catch (e) {
       log('Error caught: ', e, '\n');
-      this.error('An error occurred: ' + e);
+
+      await printError(e.toString());
+
       process.exit(-1);
     }
   }
+
+  /**
+   * Display the result.
+   *
+   * @param {SyncResult[]} result
+   */
+  async displayResult(result: SyncResult[]) {
+    const failed = result.filter(attempt => !attempt.success);
+
+    const totalCount = result.length;
+    const failedCount = failed.length;
+    const successfulCount = totalCount - failedCount;
+
+    await printLine();
+
+    if (successfulCount > 0) {
+      // Display output.
+      await printLine(`Synchronization successful for ${successfulCount} / ${totalCount} connection(s).`);
+    }
+
+    if (failed.length === 0) {
+      return process.exit(0);
+    }
+
+    // Display all errors.
+    await printLine(`Failed for following ${failedCount} connection(s):\n`);
+
+    failed.forEach(async (attempt, index) => {
+      await printLine(`${index + 1}) ${attempt.connectionId}`);
+      await printError(attempt.error);
+      await printLine();
+    });
+
+    throw new Error(`Synchronization failed for ${failedCount} / ${totalCount} connections.`);
```

Comment:
Why throw at the end? Displaying all the errors is not enough?

Reply:
We should throw the error to make sure the program exits with error code other than 0.

Reply:
Yes, or else we'll need to exit non-zero in multiple places which isn't good either.

Reply:
So `process.exit(1)`

Reply:
It's not a good idea of exit in multiple places - throwing exceptions is the right way to raise errors. We just need to ensure that there's a root handler which catches all of them.

Reply:
Cool. We already have a try catch in place. 

My only issue is why does a function that simply displays a summary or a result needs to throw any kind of error.

Reply:
There's a single place where all exceptions are caught and a non zero exit code is thrown in case of errors.

Reply:
Good point. This doesn't sound right with the display result function as it says.

Reply:
The problem is that `displayResult` did not fail but `synchronize` or something internally did. Right now display result is trying to say that something failed upstream thus I should fail. When everything is well, `displayResult` does what it is supposed to do. It prints the result and that is what it should do.

The throw should happen after `displayResult`.

```js
      const results = await synchronize(config, connections, params);

      await this.displayResult(results);

      const fail = results.find(result => result.success === false)
      
      if (fail) {
         throw new Error('...');
      }
    } catch (e) {
      // Send verbose error with stack trace to debug logs.
      log(e);

      await printError(e.toString());

      process.exit(-1);
    }
```